x-pack/apm-server/profiling/collector.go

```diff
@@ -0,0 +1,632 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package profiling
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"strconv"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/elastic/apm-server/internal/elasticsearch"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+var (
+	// metrics
+	//indexedDocs = metrics.NewLabelCounter("elastic_collector",
+	//	"indexed_document_count",
+	//	"grpc_method")
+	//indexedErrDocs = metrics.NewLabelCounter("elastic_collector",
+	//	"indexed_document_failed_count",
+	//	"grpc_method", "reason")
```

Comment:
> Would love to replace these 2 with 2 *monitoring.Ints with the same purpose.

We can, but bear in mind that `elastic-agent-libs/monitoring` does not support dimensional metrics. So we can either have 2 metrics for tracking all docs indexed/failed, or we can have a pair of metrics for each endpoint or perhaps each document type.

> What would be the best way to make this happen?

How about we get this PR merged first, and then iterate?

> My understanding is that these metrics would exposed via the Go expvar package, is that correct?

Yes. Bear in mind that expvar is disabled by default. It can be enabled, but there's no auth, and we haven't done an audit on exposure of things like command line args.

![image](https://user-images.githubusercontent.com/843579/188542068-53f5d313-ca9c-4868-97b8-bdb483fdb23b.png)

I think ideally we would be sending these metrics to telemetry.

Reply:
> How about we get this PR merged first, and then iterate?

Agreed 😄 👍🏼

> I think ideally we would be sending these metrics to the cloud monitoring cluster.

Was chatting about this yesterday with Silvia, I think we have a way of making that happen